packages/api/tests/lambdas/test-create-reconciliation-report.js

```diff
@@ -2151,3 +2152,177 @@ test.serial('Inventory reconciliation report JSON is formatted', async (t) => {
   t.true(!unformattedReportString.includes('\n')); // validate unformatted report is on a single line
   t.is(formattedReport, JSON.stringify(unformattedReportObj, undefined, 2));
 });
+
+test.serial('When there is a connection termination error for an inventory report, it retries', async (t) => {
+  const {
+    knex,
+  } = t.context;
+  const dataBuckets = [randomId('bucket')];
+  await Promise.all(dataBuckets.map((bucket) =>
+    createBucket(bucket)
+      .then(() => t.context.bucketsToCleanup.push(bucket))));
+
+  // Write the buckets config to S3
+  await storeBucketsConfigToS3(
+    dataBuckets,
+    t.context.systemBucket,
+    t.context.stackName
+  );
+
+  const queryBuilderStub = sinon.stub(QuerySearchClient.prototype, 'fetchRecords').throws(new Error('Connection terminated unexpectedly', 'PROTOCOL_CONNECTION_LOST'));
+
+  t.teardown(() => queryBuilderStub.restore());
+
+  const reportName = randomId('reportName');
+  const event = {
+    systemBucket: t.context.systemBucket,
+    stackName: t.context.stackName,
+    reportType: 'Inventory',
+    reportName,
+    startTimestamp: moment.utc().subtract(1, 'hour').format(),
+    endTimestamp: moment.utc().add(1, 'hour').format(),
+    knex,
+  };
+
+  await t.throwsAsync(
+    handler(event),
+    { message: 'Connection terminated unexpectedly' }
+  );
+  t.is(queryBuilderStub.callCount, 4);
+  sinon.assert.callCount(queryBuilderStub, 4);
+
+  const reportKey = `${t.context.stackName}/reconciliation-reports/${reportName}.json`;
+  const report = await getJsonS3Object(t.context.systemBucket, reportKey);
+  t.is(report.status, 'Failed');
+  t.is(report.reportType, 'Inventory');
+});
+
+test.serial('When there is an error for an ORCA backup report, it throws', async (t) => {
+  const dataBuckets = [randomId('bucket')];
+  await Promise.all(dataBuckets.map((bucket) =>
+    createBucket(bucket)
+      .then(() => t.context.bucketsToCleanup.push(bucket))));
+
+  // Write the buckets config to S3
+  await storeBucketsConfigToS3(
+    dataBuckets,
+    t.context.systemBucket,
+    t.context.stackName
+  );
+
+  const searchOrcaStub = sinon.stub(ORCASearchCatalogQueue.prototype, 'searchOrca');
+  searchOrcaStub.throws(new Error('ORCA error'));
+  t.teardown(() => searchOrcaStub.restore());
+
+  const reportName = randomId('reportName');
+  const event = {
+    systemBucket: t.context.systemBucket,
+    stackName: t.context.stackName,
+    reportType: 'ORCA Backup',
+    reportName,
+    startTimestamp: moment.utc().subtract(1, 'hour').format(),
+    endTimestamp: moment.utc().add(1, 'hour').format(),
+  };
+
+  await t.throwsAsync(
+    handler(event),
+    { message: 'ORCA error' }
+  );
+
+  const reportKey = `${t.context.stackName}/reconciliation-reports/${reportName}.json`;
+  const report = await getJsonS3Object(t.context.systemBucket, reportKey);
+  t.is(report.status, 'Failed');
+  t.is(report.reportType, 'ORCA Backup');
+});
+
```

Comment:
Since we move the retry logic to QuerySearchClient.ts, move/add test to QuerySearchClient

Reply:
delete these tests.

Reply:
Hm, why do you think I should delete these tests @jennyhliu ?

Reply:
Because the retry logic in now in `fetchRecords`, and we only need to test that instead.

Reply:
@jennyhliu https://github.com/nasa/cumulus/pull/3332/commits/c3679b8e447af7aebf856bb4650fff597eeca7c2#diff-f6faf60179e0607db77cd33d9b72bda73294727ed1b2e12a00e96af5c929e748L2156